File: sistema veterinario/Reptil.java
public class Reptil extends Animal
{
    public Reptil(String nombre,
                  int edad,
                  double peso,
                  Propietario propietario)
    {
        super(nombre, edad, peso, propietario);
    }

    @Override
    public void aplicarTratamiento()
    {
        System.out.println("Aplicando tratamiento para reptil.");
    }
}

File: Logica/Reptil.cs
Reptil.cs                                                         namespace Sistema_Veterinario.Logica
{
    public class Reptil : Animal
    {
        public Reptil(string nombre,
                     int edad,
                     Propietario dueño)
            : base(nombre, edad, dueño)
        {
        }

        public override void AplicarTratamiento()
        {
            Console.WriteLine(
                "Tratamiento para reptiles");
        }
    }
}